agent/csi/plugin.go

```diff
@@ -33,12 +50,133 @@ type NodePlugin struct {
 
 	// NodeClient is the node service client
 	NodeClient csi.NodeClient
+
+	// VolumeMap is the map from volume ID to Volume. Will place a volume once it is staged,
+	// remove it from the map for unstage.
+	// TODO: Make this map persistent if the swarm node goes down
+	VolumeMap map[string]*volumePublishStatus
+
+	// Lock for VolumeMap
+	Lock sync.RWMutex
+}
+
+const TargetPath string = "/var/lib/docker/%s"
+
+func NewNodePlugin(name string) *NodePlugin {
+	return &NodePlugin{
+		Name:      name,
+		VolumeMap: make(map[string]*volumePublishStatus),
+	}
+}
+
+//// Get returns a volume published path for the provided volume ID.  If the volume doesn't exist, returns empty string.
```

Comment:
* Comments should start with two slashes
* Comments immediately proceeding function definitions are godoc comments, and should start with the name of the function.

```go
// GetPublishedPath returns the path at which the provided volume ID is published.
// Returns an empty string if the volume does not exist
```